kinto/core/resource/viewset.py

```diff
@@ -226,7 +228,7 @@ def get_view_arguments(self, endpoint_type, resource_cls, method):
 
     def get_service_arguments(self):
         args = super(ShareableViewSet, self).get_service_arguments()
-        args['factory'] = authorization.RouteFactory
+        args['factory'] = self.factory
```

Comment:
Actually I think we should read that from the config.root_factory, I did a hack to get there but I wish we could fix it using the config one.